app/services/survey/withdraw.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module Survey
+  class Withdraw
+    include ServicePattern
+
+    def initialize(trainee:)
+      @trainee = trainee
+    end
+
+    def call
+      create_contact_in_mailing_list_body = create_contact_in_mailing_list_response.body
+      contact_lookup_id = JSON.parse(create_contact_in_mailing_list_body)["result"]["contactLookupId"]
+      create_distribution_response(contact_lookup_id)
+    end
+
+  private
+
+    attr_reader :trainee
+
+    delegate :first_names, :last_name, :email, :withdraw_date, :training_route, to: :trainee
+
+    def create_contact_in_mailing_list_response
+      @create_contact_in_mailing_list_response ||= QualtricsApi::Client::Request.post("/directories/#{directory_id}/mailinglists/#{mailing_list_id}/contacts", body: create_contact_in_mailing_list_body)
+    end
+
+    def create_contact_in_mailing_list_body
+      {
+        "email" => email,
+        "firstName" => first_names,
+        "lastName" => last_name,
+
+        embeddedData: {
+          "withdraw_date" => withdraw_date,
+          "training_route" => training_route,
+        },
+      }.to_json
+    end
+
+    def create_distribution_body(contact_lookup_id)
+      {
+        message: {
+          libraryId: library_id,
+          messageId: message_id,
+        },
+        recipients: {
+          mailingListId: mailing_list_id,
+          contactId: contact_lookup_id,
+        },
+        header: {
+          fromEmail: Settings.data_email,
+          fromName: "Bat Support",
+          subject: "BAT Chocolate Withdraw",
+          replyToEmail: Settings.data_email,
+        },
+        surveyLink: {
+          surveyId: survey_id,
+          expirationDate: expiration_date,
+          type: "Individual",
+        },
+        sendDate: send_date,
+        embeddedData: {
+          important: "support really needs chocolate",
+          no_clue: "guessing this can be used in email template",
+        },
+      }.to_json
+    end
+
+    def create_distribution_response(contact_lookup_id)
```

Comment:
Maybe the method name could be a bit more helpful as this step in the process is actually going to send the email out (if successful)?